src/troubleshooting/backend.rst

```diff
@@ -1,12 +1,626 @@
+    #define SSSDBG_CRIT_FAILURE   0x0020   0b00000000 00100000 /* level 1 */
+    #define SSSDBG_OP_FAILURE     0x0040   0b00000000 01000000 /* level 2 */
+    #define SSSDBG_MINOR_FAILURE  0x0080   0b00000000 10000000 /* level 3 */
+    #define SSSDBG_CONF_SETTINGS  0x0100   0b00000001 00000000 /* level 4 */
+    #define SSSDBG_FUNC_DATA      0x0200   0b00000010 00000000 /* level 5 */
+    #define SSSDBG_TRACE_FUNC     0x0400   0b00000100 00000000 /* level 6 */
+    #define SSSDBG_TRACE_LIBS     0x1000   0b00010000 00000000 /* level 7 */
+    #define SSSDBG_TRACE_INTERNAL 0x2000   0b00100000 00000000 /* level 8 */
+    #define SSSDBG_TRACE_ALL      0x4000   0b01000000 00000000 /* level 9 */
+    #define SSSDBG_BE_FO          0x8000   0b10000000 00000000 /* level 10*/
+
+                                  0xB7F0   0b10110111 11110000
+                                  0xF7F0   0b11110111 11110000
+                                  0x3FF0   0b00111111 11110000
+
+
+There are few ways to achieve this:
+
+- Using ``debug_level`` directive in domain section in ``sssd.conf``.
+
+.. code-block:: bash
+
+    # cat /etc/sssd.conf
+    ...
+    [domain/IPA.PB]
+    ...
+    debug_level = 0x3ff0
+    ...
+
+- Using ``sss_debuglevel`` tool which will set the level only for current SSSD
+  instance and will set the original level when SSSD is restarted.
+
+.. code-block:: bash
+
+    # systemctl start sssd.service
+    # sss_debuglevel 0x3ff0
+
+- Using ``sssctl`` util:
+
+.. code-block:: bash
+
+    $ sssctl debug-level <VAL>
+    $ sssctl --debug <VAL>
+
+
+Debug level ``0x3ff0`` contains enough information for all basic tasks.
+Higher level providers details on input and output operations that creates
+only noise that makes the log hard to read in most of the cases so I do not
+recommend setting it unless it is needed.
+
+Reading the logs
+----------------
+
+This section contains a description of the most fundamental parts of the
+backend and how are they described by log messages. We are trying to keep log
+messages and format as consistent as possible for all SSSD parts.
+
+Each log line contains some core information like:
+
+- Timestamp
+
+- Responder / module identifier
+
+- Function identifier
+
+- Debug level
+
+- Debug payload
+
+Sample SSSD log output example:
+
+.. code-block:: sssd-log
+
+    (Tue Nov 20 12:23:21 2020) [sssd[nss]] [cache_req_send] (0x0400): CR #59: New request 'User by name'
+    (Tue Nov 20 12:18:56 2020) [sssd[be[ldap.vm]]] [be_resolve_server_process] (0x1000): Saving the first resolved server
+    (Tue Nov 20 12:18:56 2020) [sssd[be[ldap.vm]]] [be_resolve_server_process] (0x0200): Found address for server master.ldap.vm: [192.168.100.20] TTL 0
+    (Tue Nov 20 12:18:56 2020) [sssd[be[ldap.vm]]] [sdap_uri_callback] (0x0400): Constructed uri 'ldap://master.ldap.vm:389'
+
+In the following chapters we will skip first part of the log line which is timestamp
+for the more clear log snippets.
```

Comment:
`In the following chapters we will exclude the first column timestamp in the log snippets` sounds a bit more clear IMO

Reply:
Fixed